### PR TITLE
Fix hooks exports. Remove unneeded deps. Update deps.

### PR DESCRIPTION
**Pull request by kaisermann**
- [x] Fix `@testing-library/...` exports
- [x] Kept exported `renderHook` method, but also added a `hooks` bag of methods because the default `act()` doesn't work properly with hooks, so the user must use `hooks.act()`
- [x] Added our prettier and eslint configs
- [x] Removed unneeded dependencies
- [x] Removed some unused code
- [x] Fix typings for extended dom matchers

### CODE REVIEW

File: react.js [hunk 1]
```diff
@@ -1,26 +1,23 @@
-/* eslint-disable global-require */
-
 const React = require('react')
 const reactTestingLibrary = require('@testing-library/react')
-const reactHooksTestingLibrary = require('@testing-library/react-hooks')
+const hooks = require('@testing-library/react-hooks')
 const { MockedProvider } = require('@apollo/react-testing')
 const { IntlProvider } = require('react-intl')
-const { path, find, isNil, reject } = require('ramda')
```
Comment (kaisermann) [inline react.js:8]: muahahaha

File: react.js [hunk 2]
```diff
@@ -1,26 +1,23 @@
...
 const { InMemoryCache } = require('apollo-cache-inmemory')
 
 const paths = require('./modules/paths')
 
 const pkg = require(paths.resolveAppPath('package.json'))
 
 const getLocale = (optionsLocale) => {
-  const pkgLocale = path(['vtexTestTools', 'defaultLocale'], pkg)
+  const pkgLocale = pkg.vtexTestTools.defaultLocale
```
Comment (rafarubim) [inline react.js:13]: Any null reference risk?

Comment (kaisermann) [reply react.js:13]: The previous code wasn't checking for any fallback value 🤷 And we're testing for its truthiness two lines below!

Comment (kaisermann) [reply react.js:13]: But I see, `vtexTestTools` could be undefined, will add a check for that!

Comment (rafarubim) [reply react.js]: What about a thrown error? Just to be sure 😬

File: react.js [hunk 3]
```diff
@@ -1,26 +1,23 @@
...
```
Comment (klzns) [inline react.js]: It's it possible to have an error for:
> `defaultLocale` of undefined

Comment (klzns) [reply react.js]: Outdated

**Pull request merged**